torch/distributed/elastic/rendezvous/dynamic_rendezvous.py

```diff
@@ -169,7 +169,7 @@ class RendezvousSettings:
     keep_alive_max_attempt: int
 
 
-@dataclass(eq=True, frozen=True)
+@dataclass(eq=True, order=True, frozen=True)
 class _NodeDesc:
     """Describes a node in the rendezvous.
```

Comment:
doesn't `dataclass.order` order w.r.t the parameter list order? so in this case (on the same machine) `NodeDesc(host_a, pid=100, local_id=1)` is less than `NodeDesc(host_a, pid=101, local_id=0)`? I guess it depends on where you want to use this comparator, but seems a bit weird to order by pid first no? (we typically order by rank).

Side note: if `local_id` == `local_rank` can we name it as `local_rank` instead?

Reply:
`local_id` does not map to `local_rank`. The only reason of having a `local_id` attribute in `_NodeDesc` is to distinguish between two `DynamicRendezvousHandler` instances in the same process belonging to the same distributed job. Although this kind of set up is unlikely to happen in a production environment, it is convenient for integration tests to simulate a distributed job inside a single process.